server/files/handler.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from concurrent.futures import Future
+
+from ..actor import Actor
+from ..comm.protocol import OnFileUpload, FileUpload
+
+
+# NOTE(bgrzesik): Maybe changing this to Actor would solve blocking
+# other uploads
+class UploadedFile(object):
+
+    def __init__(self, name, rw=False):
+        logging.debug(f"UploadedFile(): name={name} rw={rw}")
+        self._name = name
+        self._file = open(name, "wb" if rw else "r")
+        self._next_part = 0
+
+    def append_part(self, part: FileUpload.Part) -> FileUpload.Result:
+        logging.debug(f"append_part(): part_no={part.part_no} "
+                      f"size={len(part.chunk)}")
+
+        if self._next_part != part.part_no:
+            return FileUpload.Result.IO_ERROR
+
+        self._next_part += 1
+        self._file.write(part.chunk)
+        self._file.flush()
+
+        return FileUpload.Result.OK
+
+    def finish_upload(self, part: FileUpload.Finish) -> FileUpload.Result:
+        logging.debug(f"finish_upload(): name={self._name}")
+
+        self._file.close()
+        self._file = open(self._name, "r")
+
+        return FileUpload.Result.OK
+
+
+class FileStore(Actor):
+
+    def __init__(self):
+        Actor.__init__(self)
+        self._store = {}
+        self._next_upload_id = 0
+        self._uploads = {}
+
+    @Actor.assert_executor()
+    def start_upload(self, name):
+        logging.debug(f"start_upload(): name={name}")
+        new_file = UploadedFile(name, rw=True)
```

Comment:
path prefix should be added

Reply:
I agree and additional checks, but I wanted to push out those changes as soon as possible.

Reply:
Let’s merge and add this with configuration